pkg/prometheus/rules.go

```diff
@@ -332,20 +332,37 @@ func prometheusRuleConfigMapName(prometheusName string) string {
 }
 
 // GenerateContent takes a PrometheusRuleSpec and generates the rule content
-func GenerateContent(promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
+func GenerateContent(objectType string, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
```

Comment:
2 functions would be more explicit IMHO. I'd also recommend moving the code to the pkg/operator package since it's shared between prometheus and thanos.

```suggestion
func GenerateRulesConfigurationForPrometheus(promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
    // Unset partialResponseStrategy field.
    for _, for i := range promRule.Groups {
        ...
    }
    return generateRulesConfiguration(promRule, logger)
}

func GenerateRulesConfigurationForThanos(objectType string, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
    return generateRulesConfiguration(promRule, logger)
}

func generateRulesConfiguration(objectType string, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
    // same as GenerateContent
}
```

Reply:
Thanks for that suggestion :) It looks much better :)